sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkTracer.java

```diff
@@ -13,17 +13,23 @@
 /** {@link SdkTracer} is SDK implementation of {@link Tracer}. */
 final class SdkTracer implements Tracer {
   static final String FALLBACK_SPAN_NAME = "<unspecified span name>";
+  private static final Tracer NOOP_TRACER = TracerProvider.noop().get("noop");
 
   private final TracerSharedState sharedState;
   private final InstrumentationScopeInfo instrumentationScopeInfo;
+  private final TracerConfig tracerConfig;
 
   SdkTracer(TracerSharedState sharedState, InstrumentationScopeInfo instrumentationScopeInfo) {
     this.sharedState = sharedState;
     this.instrumentationScopeInfo = instrumentationScopeInfo;
+    this.tracerConfig = sharedState.getTracerConfig(instrumentationScopeInfo);
   }
 
   @Override
   public SpanBuilder spanBuilder(String spanName) {
+    if (!tracerConfig.isEnabled()) {
+      return NOOP_TRACER.spanBuilder(spanName);
```

Comment:
If I understand correctly, noop tracer returns invalid span which have some side effects like this:

```java

Span parent = tracerDisabledScope.spanBuilder("foo") // suppressed
try (Scope s = span.makeCurrent()) {
   // span.Current is invalid here, but 
   Span child = tracerEnabledScope.spanBuilder("bar") // not suppressed, but context is lost
   ...
}
```
In this case `makeCurrent` should probably make parentContext current

it'd be great to allow instrumentation to write their code without thinking about suppression (they should be able to know about it if they want to optimize).

Reply:
> In this case makeCurrent should probably make parentContext current

This should be the case. See the implementation [here](https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/trace/DefaultTracer.java).

Note how [startSpan](https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/trace/DefaultTracer.java#L47) returns a span wrapping the parent context (unless `SpanBuilder#setParent` explicitly overrides this).

Reply:
make sense, thanks!